pkg/admission/validate_noobaa.go

```diff
@@ -40,8 +45,24 @@ func (nv *ResourceValidator) ValidateNoobaa() admissionv1.AdmissionReview {
 	return *nv.arResponse
 }
 
+// DeserializeNB extract noobaa cr from the request
+func (nv *ResourceValidator) DeserializeNB(rawNB []byte) *nbv1.NooBaa {
+	NB := nbv1.NooBaa{}
+	if err := json.Unmarshal(rawNB, &NB); err != nil {
+		nv.Logger.Error("error deserializing noobaa")
+	}
+	return &NB
+}
+
 // ValidateDeleteNoobaa runs all the validations tests for DELETE operations
 func (nv *ResourceValidator) ValidateDeleteNoobaa() {
-	nv.Logger.Infof("Validating DELETE operation")
-	nv.SetValidationResult(false, "Deletion of NooBaa resource is prohibited")
+	noobaaCR := nv.DeserializeNB(nv.arRequest.Request.OldObject.Raw)
+	if noobaaCR == nil {
```

Comment:
what does it mean that noobaaCR == nil?

Reply:
looking at `DeserializeNB`, if I'm not mistaken, noobaaCR cannot be nil.
I think that since we are dealing with a destructive operation, we might want to deny on every error we get